### PR TITLE
wifi: SAP mode docs

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: doc/nrf/glossary.rst
Comment (greg-fer) [inline doc/nrf/glossary.rst:685]:
```suggestion
      A network device operating as an IEEE 802.11 access point using software with limited functionality and no connection to the wired network.
```

**Review by greg-fer — COMMENTED** [review]
Checked glossary.rst. Will hold off on approval until @richabp reviews the docs.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:17]: Do we need to state this here ? I guess this should be updated in https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13562/nrf/releases_and_maturity/software_maturity.html#wi-fi-feature-support ?

Comment (greg-fer) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:17]: To avoid redundancy, I suggest at least mentioning "The SAP mode support is experimental" and linking to software_maturity.rst under "experimental", as we do in other places.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:26]: MAybe you can add a more strict sentence. E.g.
The SAP mode current supports up to a single Station device connection

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:30]:
```suggestion
Enabling the SAP mode
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:33]:
```suggestion
The SAP mode is enabled by setting the :kconfig:option:`CONFIG_NRF700X_AP_MODE` and :kconfig:option:`CONFIG_WPA_SUPP_AP` options in the application project configuration file.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:39]: IS there a handy link to that? Would be nice to have it here :)

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:105]:
```suggestion
    The SAP mode operation is dictated by regulatory requirements.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:114]:
```suggestion
Currently the Zephyr networking stack does not support DHCP server functionality.
```
Also, could this be made into a note?

**Review by ioannisg — COMMENTED** [review]
@krish2718  some early doc comments and suggestions from my side!

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:17]: ok, I forgot to tag this support as `EXPERIMENTAL` will do it in this PR.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:24]: Zephyr only has a single interface type, I can change to `network interface`

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:25]: We are not providing any configuration, it *only* operates in `11n`.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:26]: Supports up to 8, but tested only with 1, so, may be: `SAP mode support up to eight clients, but the testing has been carried out with a single client`?

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:28]: ok, will remove

File: doc/nrf/device_guides/working_with_nrf/nrf70/features.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/features.rst:86]: I think we should just use the acronym that was introduced in the `glossary`.

**Review by sachinthegreen — APPROVED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:14]:
```suggestion
The smartphone acts as a virtual router, allowing other station devices to connect to the internet through it.
SoftAP mode may, alternatively, be used to securely provision Wi-Fi devices into an access point network.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:16]:
```suggestion
nRF70 Series devices support Wi-Fi provisioning using the SoftAP mode.
```

Comment (ioannisg) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:16]: Note to @greg-fer : I hope the sentence above is not too "strong", i.e. it does not imply that SoftAP-based provisioning is the _only_ way to provision nRF70 Series devices.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:24]: OK

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:24]:
```suggestion
  * The SAP mode supports only Wi-Fi 4 (IEEE 802.11n) mode of operation.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (ioannisg) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:25]: How about:

```suggestion
  * The SAP mode currently supports a single station device connection. The implementation provides _experimental_ support for up to 8 station connection is added.
```

File: doc/nrf/glossary.rst
Comment (ioannisg) [inline doc/nrf/glossary.rst:685]: Now, I am thinking, this contradicts a bit with the definition of SoftAP in the guide (... as a means of sharing an internet connection). Can we drop the last part?

```suggestion
      A network device operating as an IEEE 802.11 access point using software with limited functionality.
```

**Review by ioannisg — COMMENTED** [review]
some more comments @krish2718 to improve messaging.

I would completely drop the "SAP" abbreviation, or make it clear inside the guide that it can be used interchangeably with "SoftAP". But I leave you and @greg-fer to decide.

**Comment by krish2718** [comment]
> I would completely drop the "SAP" abbreviation, or make it clear inside the guide that it can be used interchangeably with "SoftAP". But I leave you and @greg-fer to decide.

We have put both the terms in the `glossary`, so, should be fine.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst]: I have added `as one of the provisioning methods.`

File: doc/nrf/glossary.rst
Comment (ioannisg) [inline doc/nrf/glossary.rst:684]:
```suggestion
   Software-Enabled Access Point (SoftAP or SAP)
```

**Review by ioannisg — APPROVED** [review]
Looks good @krish2718 , thanks! Possible minor typo in glossary spotted.

File: doc/nrf/device_guides/working_with_nrf/nrf70/features.rst
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/features.rst:86]: IMO, using :term: is more relevant than :abbr:.
```suggestion
Currently, :term:`Station (STA)` and :term:`Software-enabled Access Point (SoftAP or SAP)` modes are supported.
```

File: samples/wifi/shell/README.rst
Comment (richabp) [inline samples/wifi/shell/README.rst:304]:
```suggestion
To test the SAP mode, the sample must be built using the configuration overlay :file:`overlay-sap.conf` file.
```

File: samples/wifi/shell/README.rst
Comment (richabp) [inline samples/wifi/shell/README.rst:322]:
```suggestion
#. Set an IP address for the SAP interface using the following command:
```

File: samples/wifi/shell/README.rst
Comment (richabp) [inline samples/wifi/shell/README.rst:361]:
```suggestion
#. Connect a station to the SAP using a static IP address and verify the connection using the following command:
```

File: samples/wifi/shell/README.rst
Comment (richabp) [inline samples/wifi/shell/README.rst:348]:
```suggestion
#. Check the SAP status after a while, using the following command:

   .. code-block:: console

       wifi status

   If the SAP is established, you should see an output similar to the following:

   .. code-block:: console

        Status: successful
        ==================
        State: COMPLETED
```

File: samples/wifi/shell/README.rst
Comment (richabp) [inline samples/wifi/shell/README.rst:367]: Formatting issue, line 367 to line 377 are displayed in the previous code block.

**Review by richabp — CHANGES_REQUESTED** [review]
Needs a Changelog entry for SAP documentation.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:103]: But both are different topics, let's leave it as is.

**Review by ioannisg — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf70/features.rst
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/features.rst:86]:
```suggestion
Currently, :term:`Station mode (STA)` and :term:`Software-enabled Access Point (SoftAP or SAP)` mode are supported.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:30]: Formatting issue https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13562/nrf/device_guides/working_with_nrf/nrf70/developing/sap.html#supported-functionality-and-limitations
```suggestion
  * A network interface can operate in either SAP mode or Station mode, but not in both modes simultaneously.
  * SAP mode only supports the Wi-Fi 4 (IEEE 802.11n) mode of operation.
  * The SAP mode currently supports a single station device connection.
     However, the implementation provides an :ref:`experimental <software_maturity>` support for up to eight station connections.
 * Only Wi-Fi personal security (WPA2-PSK) is supported.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (richabp) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:107]: May be then it could be written as a plain text?

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]
Approving for the _glossary_ update only. Documentation update still requires +1 from @richabp .

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:30]: if I don't put a newline, the previous line is highlighted in `bold`

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst
Comment (krish2718) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst:107]: No, but this is important to highlight.

**Review by richabp — APPROVED** [review]

**Pull request merged**
